utils/firestore.js

```diff
@@ -1369,16 +1463,21 @@ const getLocations = async (institute) => {
 
 const searchBoxes = async (institute, flag) => {
     let snapshot = ``
-    if ((institute === nciCode || institute == nciConceptId) && flag === `bptl`) {
-        snapshot = await db.collection('boxes').get()
-    } 
-    else { 
+    if (institute === nciCode || institute == nciConceptId) {
+        if (flag === `bptl`) {
+            snapshot = await db.collection('boxes').get();
+        } else if (flag === `bptlPackagesInTransit`) {
+            snapshot = await db.collection("boxes")
+                            .where(fieldMapping.submitShipmentFlag.toString(), "==", fieldMapping.yes)
+                            .where(fieldMapping.siteShipmentReceived.toString(), "==", fieldMapping.no).get();
+        }
```

Comment:
nit: for defensiveness we can either include else { (throw error flag is missing) } here OR check for flag in the initial `if` clause. Neither case is expected, but it alerts us if there's a missing flag in one of our API calls coming from BSD:
```javascript
if (flag && (institute === nciCode || institute == nciConceptId)) {
```

Reply:
I was initially going to add your suggested else block to throw an error, but adding the flag in the initial `if` conditional to check for the existence of the flag parameter makes reading the code easier. Thanks for both suggestions.